### DISCUSSION TITLE
CORS configuration isnt working...Pls help

### DISCUSSION THREAD

**Discussion by manjug2021** [DependencyTrack/dependency-track#968] [Q&A] [answered]
Hello,

We have setup api and frontend containers on aws on two separate EC2 instances. We have used docker-compose.yml to configure both api and frontend containers. While frontend is able to be accessible but login doesnt seem to work properly. Browser logs indicate CORS issue. API however is able to communicate to requests from POSTMAN without any issue.

Any one knows how to resolve this issue. Here is configuration from docker-compose.yml file for API.

`version: '3.7'

#####################################################
# This Docker Compose file contains following services
#    Dependency-Track API Server
#####################################################

volumes:
  dependency-track:

services:
  dtrack-apiserver:
    image: dependencytrack/apiserver
    environment:
    # The Dependency-Track container can be configured using any of the
    # available configuration properties defined in:
    # https://docs.dependencytrack.org/getting-started/configuration/
    # All properties are upper case with periods replaced by underscores.
    #
    # Database Properties
     - ALPINE_DATABASE_MODE=external
     - ALPINE_DATABASE_URL=jdbc:postgresql://10.0.1.25:5432/dtrack
     - ALPINE_DATABASE_DRIVER=org.postgresql.Driver
     - ALPINE_DATABASE_USERNAME=dtrackuser
     - ALPINE_DATABASE_PASSWORD=################
     - ALPINE_DATABASE_POOL_ENABLED=true
     - ALPINE_DATABASE_POOL_MAX_SIZE=10
     - ALPINE_DATABASE_POOL_IDLE_TIMEOUT=600000
     - ALPINE_DATABASE_POOL_MAX_LIFETIME=600000

    # Optional Cross-Origin Resource Sharing (CORS) Headers
     - ALPINE_CORS_ENABLED=true
     - ALPINE_CORS_ALLOW_ORIGIN=*
     - ALPINE_CORS_ALLOW_METHODS=GET POST PUT DELETE OPTIONS
     - ALPINE_CORS_ALLOW_HEADERS=Origin, Content-Type, Authorization, X-Requested-With, Content-Length, Accept, Origin, X-Api-Key, X-Total-Count, *
     - ALPINE_CORS_EXPOSE_HEADERS=Origin, Content-Type, Authorization, X-Requested-With, Content-Length, Accept, Origin, X-Api-Key, X-Total-Count
     - ALPINE_CORS_ALLOW_CREDENTIALS=true
     - ALPINE_CORS_MAX_AGE=3600
    deploy:
      resources:
        limits:
          memory: 6144m
        reservations:
          memory: 4608m
      restart_policy:
        condition: on-failure
    ports:
      - '8080:8080'
    volumes:
      - 'dependency-track:/data'
    restart: unless-stopped`

**Comment by nscuro** [comment, marked as answer]
Change

```
ALPINE_CORS_ALLOW_METHODS=GET POST PUT DELETE OPTIONS
```

to 

```
ALPINE_CORS_ALLOW_METHODS=GET,POST,PUT,DELETE,OPTIONS
```

See https://developer.mozilla.org/en-US/docs/Web/HTTP/Headers/Access-Control-Allow-Methods#directives

**Reply by manjug2021** [reply]
Yes it works. Thanks for your help.
